homeassistant/components/ista_ecotrend/coordinator.py

```diff
@@ -0,0 +1,90 @@
+"""DataUpdateCoordinator for Ista EcoTrend integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from pyecotrend_ista.exception_classes import (
+    InternalServerError,
+    KeycloakError,
+    LoginError,
+    ServerError,
+)
+from pyecotrend_ista.pyecotrend_ista import PyEcotrendIsta
+from requests.exceptions import RequestException
+
+from homeassistant.const import CONF_EMAIL
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IstaCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Ista EcoTrend data update coordinator."""
+
+    details: dict[str, Any]
+
+    def __init__(self, hass: HomeAssistant, ista: PyEcotrendIsta) -> None:
+        """Initialize ista EcoTrend data update coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(days=1),
+        )
+        self.ista = ista
+        self.details = {}
```

Comment:
```suggestion

    def __init__(self, hass: HomeAssistant, ista: PyEcotrendIsta) -> None:
        """Initialize ista EcoTrend data update coordinator."""
        super().__init__(
            hass,
            _LOGGER,
            name=DOMAIN,
            update_interval=timedelta(days=1),
        )
        self.ista = ista
        self.details: dict[str, Any] = {}
```
or is there a specific reason I missed, why this should be indeed a class variable?